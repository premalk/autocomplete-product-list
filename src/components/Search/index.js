import { func } from 'prop-types';
import styled from 'styled-components';

const Input = styled.input`
  width: 70%;
  padding: 1.5em;
  margin: 1.5em;
  border: none;
  border-radius: 3px;
`;

const Search = ({ searchCallback }) => {
  const onChange = (e) => {
    searchCallback(e.target.value);
  };

  return <Input type="text" placeholder="Search" onChange={onChange} />;
};

Search.propTypes = {
  searchCallback: func
};

export default Search;
