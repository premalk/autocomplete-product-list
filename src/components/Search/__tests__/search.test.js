import { render, act, fireEvent } from '@testing-library/react';
import Search from '../index';

describe('Testing Search', () => {
  const search = render(<Search />);

  test('Snapshot Test of search', () => {
    expect(search).toMatchSnapshot();
  });
  it('Pagination button testing', () => {
    const mockFn = jest.fn();
    const { getByTestId, rerender } = render(<Search callback={mockFn} />);

    act(() => {
      fireEvent.change(getByTestId('filter-search'), {
        target: { value: 'search' }
      });
    });

    rerender(<Search callback={mockFn} />);

    expect(getByTestId('filter-search').value).toBe('search');
    expect(mockFn).toBeCalledTimes(1);
  });
});
