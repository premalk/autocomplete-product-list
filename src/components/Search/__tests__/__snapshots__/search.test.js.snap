// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Search Snapshot Test of search 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <input
        class="styled-input"
        data-testid="filter-search"
        id="search"
        placeholder="Search"
        type="text"
      />
    </div>
  </body>,
  "container": <div>
    <input
      class="styled-input"
      data-testid="filter-search"
      id="search"
      placeholder="Search"
      type="text"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
